shared_libraries/orca_shared/recovery/test/unit_tests/test_shared_recovery.py

```diff
@@ -64,22 +58,18 @@ def test_post_entry_to_fifo_queue_no_errors(self):
         function returns the same expected message.
         """
         new_data = {"name": "test"}
-        for request_method in self.request_methods:
-            # Run subtests
-            with self.subTest(request_method=request_method):
-                # Send values to the function
-                shared_recovery.post_entry_to_fifo_queue(
-                    new_data, request_method, self.db_queue_url
-                )
+        # Run subtests
+        # Send values to the function
```

Comment:
So originally we tested both 
```
self.request_methods = [
            shared_recovery.RequestMethod.NEW_JOB,
            shared_recovery.RequestMethod.UPDATE_FILE,
        ]
```

Any reason why we are only testing `NEW_JOB` now?

Reply:
Unfortunately FIFO queue does not work with subtests. If we use the same input for different tests using  `NEW_JOB` and `UPDATE_FILE `, it throws an error. 

```
 IndexError: list index out of range
```

My assumption is since we are sending the same input to FIFO queue, it only accepts the first message and ignores the one ones.

Reply:
So I did a little digging and figured out what was going on. So the updates to moto actually put it closer in reality to what an SQS queue does. So in order to better test, here are the modifications I made to the original test file. Note that I am only showing the additions and a few reference lines, not the whole code.

```python
def setUp(self):
    """
    Perform initial setup for the tests.
    """
    ...
    self.fifo_queue = self.test_sqs.create_queue(
        QueueName="unit-test-fifo-queue.fifo", Attributes={"FifoQueue": "true"}
    )
    self.fifo_queue_url = self.fifo_queue.url
    ...

def test_post_entry_to_fifo_queue_no_errors(self):
        """
        *Happy Path*
        Test that sending a message to SQS queue using post_entry_to_fifo_queue()
        function returns the same expected message.
        """
        new_data = {"name": "test"}
        for request_method in self.request_methods:
            with self.subTest(request_method=request_method):
                # Send values to the function
                shared_recovery.post_entry_to_fifo_queue(
                    new_data, request_method, self.fifo_queue_url
                )
                queue_contents = self.fifo_queue.receive_messages(
                    MessageAttributeNames=["All"]
                )
                
                # ADDING test to verify message attribute
                self.assertEqual(
                    request_method.value,
                    queue_contents[0].message_attributes["RequestMethod"][
                        "StringValue"
                    ],
                )
                queue_output_body = json.loads(queue_contents[0].body)

                # Testing SQS body
                self.assertEqual(queue_output_body, new_data)
                
                # DELETE the message from the queue to show we have read it
                self.fifo_queue.delete_messages(
                    Entries=[
                        {
                            "ReceiptHandle": queue_contents[0].receipt_handle,
                            "Id": queue_contents[0].message_id,
                        }
                    ]
                )

```

Reply:
This works! I forgot that the message needs to be read first in FIFO before going to the next message. Good to know moto is smart to handle that!